### PR TITLE
changed cowpoke to use account auth

### PR DESCRIPTION
**Pull request by EvanSnapp**

### CODE REVIEW

File: src/rancher.js [hunk 1]
```diff
@@ -85,8 +68,15 @@ function listStacks( http, stackUrl) {
 	}, error => error );
 }
 
+function listEnvironments(http) {
+	return http.get("/v1/projects").then(res => res.data.map(elm => {
+		elm.listStacks = listStacks.bind( null, http, elm.links.environments, elm.name );
+		elm.listContainers = http.get.bind( null, elm.links.containers );
```
Comment (shawnHartsell) [inline src/rancher.js]: I could not find any code that used `listContainers`. Is it needed?

File: resource/stack/stack.js [hunk 2]
```diff
@@ -77,54 +69,33 @@ const upgradeStack = Promise.coroutine(function* ( slack, envelope ) {
...
-			secret: storedEnvironments[i].secret
-		} ).catch(() => console.error("Authorization failed for ", storedEnvironments[i].name)).then(Promise.coroutine(function*(rancher) {
-			if (rancher === undefined) {
-				throw new Error("Authorization failed for " + storedEnvironments[i].name);
-			}
-			const environments = yield rancher.listEnvironments();
-			return environments[_.keys( environments )[0]];
-		})));
+	const enviorments = yield rancherFn(rancherUrl, rancherUser).then(elm => elm.listEnvironments()).catch(() => undefined);
+	if ( enviorments === undefined ) {
```
Comment (shawnHartsell) [inline resource/stack/stack.js:73]: the code should handle situations where the request went through, but the user did not have access to any rancher environments (e.g. empty array). This case would probably warrant a different kind of HTTP code as well (403, etc)

**Comment by shawnHartsell** [comment]
In addition to the comments above, the code not passing `gulp lint`

File: src/rancher.js (hunk 1 above)
Comment (EvanSnapp) [reply src/rancher.js]: You are right! That is dead code that I do not think has ever been used and I forgot to remove it.

**Pull request merged**